indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -4220,6 +4220,16 @@ protected void emitLag()
           return;
         }
 
+        // Try emitting lag even with stale metrics provided that none of the partitions has negative lag
+        final boolean areOffsetsStale = sequenceLastUpdated != null
+                                        && sequenceLastUpdated.isBefore(
+            System.currentTimeMillis() - tuningConfig.getOffsetFetchPeriod().getMillis()
+        );
+        if (areOffsetsStale && partitionLags.values().stream().anyMatch(x -> x < 0)) {
+          log.warn("Skipping negative lag emission as fetched offsets are stale");
```

Comment:
lets rephrase it in a way that is more informative for someone reading this. 
Lag is negative and will not be emitted because topic offsets have become stale. This will not impact data processing. Offsets become stale because....